### ISSUE TITLE
[bug] Type facet capitalization issues; investigate one-off script options to normalize in OpenSearch

### ISSUE THREAD

**Issue by christinklez** [ucldc/rikolti#863]
See -stage for the current list of type facet options: https://calisphere-stage.cdlib.org/search/?q=

**Comment by amywieliczka** [comment]
Option 1:
- In `calisphere_solr.calisphere_solr`, replace https://github.com/ucldc/rikolti/blob/main/metadata_mapper/mappers/calisphere_solr/calisphere_solr_mapper.py#L40 `type: self.source_metadata.get("type", None)` with `type: [lower(t) for t in self.source_metadata.get('type', None)]`
- In `solr_updater`, replace https://github.com/ucldc/rikolti/blob/466f1c0516d4a71a23fb9309628bedc62d137812/metadata_mapper/mappers/mapper.py#L1496 `'type': filter_blank_values(record.get('type'))` with `'type': [lower(t) for t in filter_blank_values(record.get('type'))]`
- Write a one-off script for reading every record in OpenSearch and updating the type field to be lowercase

This wouldn't hold up harvesting - if harvesting continues, we will use the one-off script to fix more records than currently exist, but we have records that need fixing anyway. 

One implication, though: in the legacy harvester, we had a "raw" type value (`type`), as well as a normalized, lowercase type value (`type_ss`). As far as @barbarahui and I can tell, the "raw" `type` value was never used anywhere except for in the item level metadata display. We _may_ lose some instances of capitalized type for some records if we choose to normalize the `type` field. It's really impossible to figure out how widespread instances of a capitalized "raw" type value - many mappers already normalize type to a lowercase value anyway. We were able to find this one instance, though: https://calisphere.org/item/9a25f3bd3e1d97e88869404d5b497e67/. 

Option 2: We could perform all the same updates as above, and also add a new type field to our opensearch schema to store the type value in whatever format it is in when it exits the mapper, and then direct Calisphere's metadata display to use this secondary, new type field in the one case where we display this maybe-capitalized type value. 

We really like Option 1. Given the fact that we are already normalizing Type extensively (`Photographic Print` to `image`, for example), normalizing `Text` to `text` does not seem like a significant metadata quality loss. 




Next steps are to investigate how quickly and easily we can write a script to perform a one-off update to fix records already in place in the index, and if there are any issues with this tactic. Once we have determined that, we can start harvesting in earnest again. If there are issues, though, we may have to hold on harvesting until the fixes are in place and also re-harvest everything.

**Comment by amywieliczka** [comment]
Chad sent us a tool awhile back for updating OpenSearch records. Besides the technical tooling, we need to make some determinations regarding whether or not it's an issue to have OpenSearch records that are out-of-sync with s3 records.

**Comment by barbarahui** [comment]
Mapper updates implemented in this PR: https://github.com/ucldc/rikolti/pull/884

These are the tools that Chad found that might be useful for updating opensearch records:

https://github.com/elasticsearch-dump/elasticsearch-dump

https://opensearch.org/docs/latest/im-plugin/reindex-data/

**Comment by barbarahui** [comment]
I used the "update by query" API to update existing records with uppercase `Title` and `Image` type.

I ended up simply issuing this API call in the OpenSearch dashboard since it was a one-off thing:

```
POST rikolti-stg/_update_by_query
{
  "query": {
    "term": {
      "type.raw": "Text"
    }
  },
  "script": {
    "source": "ctx._source.type = ctx._source.type[0].toLowerCase()",
    "lang": "painless"
  }
}
```

That updated the records with type of `Text`. Then I modified the query for `Image` and ran it again.

@amywieliczka @bibliotechy This wasn't that hard to figure out, but probably good to have this saved somewhere for future reference. I'll add it to the rikolti/record_indexer README:  https://github.com/ucldc/rikolti/blob/main/record_indexer/README.md
